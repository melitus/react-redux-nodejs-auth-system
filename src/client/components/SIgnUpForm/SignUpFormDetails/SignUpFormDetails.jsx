// @flow
import React from 'react';
import { Field, FormSection } from 'redux-form';
import { Icon } from 'semantic-ui-react';

// import { LabelInputField } from 'react-semantic-redux-form';
import { LabelInputField } from '../../Field/index';

type SignUpFormDetailsProp = {
  firstName?: string,
  lastName?: string,
  password?: number,
  email: string,
  phoneNumber: number,
};

const SignUpFormDetails = ({ firstName, lastName, password, email, phoneNumber } : SignUpFormDetailsProp) => (
  <FormSection name="signUpFormDetails">
    <label target="signup.signUpFormDetails.firstName">First Name</label>
    <Field
      id="signup.signUpFormDetails.firstName"
      name="firstName"
      component={LabelInputField}
      label={{ content: <Icon color="blue" name="user" size="small" /> }}
      labelPosition="left"
      placeholder="First Name"
      value={firstName}
      autoFocus
    />

    <label target="signup.signUpFormDetails.lastName">Last Name</label>
    <Field
      id="signup.signUpFormDetails.lastName"
      name="lastName"
      component={LabelInputField}
      label={{ content: <Icon color="blue" name="user" size="small" /> }}
      labelPosition="left"
      placeholder="Last Name"
      value={lastName}
    />

    <label target="signup.signUpFormDetails.password">Password</label>
    <Field
      id="signup.signUpFormDetails.password"
      name="Password"
      component={LabelInputField}
      label={{ content: <Icon color="blue" name="lock" size="small" /> }}
      labelPosition="left"
      placeholder="••••••••••"
      type="password"
      value={password}
    />

    <label target="signup.signUpFormDetails.email">Email</label>
    <Field
      id="signup.signUpFormDetails.email"
      name="email"
      component={LabelInputField}
      label={{ content: <Icon color="blue" name="envelope" size="small" /> }}
      labelPosition="left"
      placeholder="Email"
      type="email"
      value={email}
    />

    <label target="signup.signUpFormDetails.phoneNumber">Phone Number</label>
    <Field
      id="signup.signUpFormDetails.phoneNumber"
      name="phoneNumber"
      component={LabelInputField}
      label={{ content: <Icon color="blue" name="phone" size="small" /> }}
      labelPosition="left"
      placeholder="Phone Number"
      type="number"
      value={phoneNumber}
    />
  </FormSection>
);

SignUpFormDetails.defaultProps = {
  firstname: '',
  lastname: '',
  password: '',
  email: '',
  phoneNumber: '',
};

export default SignUpFormDetails;
