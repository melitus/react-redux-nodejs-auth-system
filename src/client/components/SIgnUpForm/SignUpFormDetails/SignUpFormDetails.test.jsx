import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import SignUpFormDetails from './SignUpFormDetails';

enzyme.configure({ adapter: new Adapter() });

describe('<SignUpFormDetails />', () => {
  it('should render default snapshot', () => {
    const wrapper = shallow(
      <SignUpFormDetails />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
