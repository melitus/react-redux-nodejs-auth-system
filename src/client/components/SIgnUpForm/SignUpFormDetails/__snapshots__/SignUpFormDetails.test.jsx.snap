exports[`<SignUpFormDetails /> should render default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpFormDetails
    email=""
    firstname=""
    lastname=""
    password=""
    phoneNumber="" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label
          target="signup.signUpFormDetails.firstName">
          First Name
</label>,
        <Field
          autoFocus={true}
          component={[Function]}
          id="signup.signUpFormDetails.firstName"
          label={
                    Object {
                              "content": <Icon
                                as="i"
                                color="blue"
                                name="user"
                                size="small" />,
                            }
          }
          labelPosition="left"
          name="firstName"
          placeholder="First Name"
          value={undefined} />,
        <label
          target="signup.signUpFormDetails.lastName">
          Last Name
</label>,
        <Field
          component={[Function]}
          id="signup.signUpFormDetails.lastName"
          label={
                    Object {
                              "content": <Icon
                                as="i"
                                color="blue"
                                name="user"
                                size="small" />,
                            }
          }
          labelPosition="left"
          name="lastName"
          placeholder="Last Name"
          value={undefined} />,
        <label
          target="signup.signUpFormDetails.password">
          Password
</label>,
        <Field
          component={[Function]}
          id="signup.signUpFormDetails.password"
          label={
                    Object {
                              "content": <Icon
                                as="i"
                                color="blue"
                                name="lock"
                                size="small" />,
                            }
          }
          labelPosition="left"
          name="Password"
          placeholder="Password"
          type="password"
          value="" />,
        <label
          target="signup.signUpFormDetails.email">
          Email
</label>,
        <Field
          component={[Function]}
          id="signup.signUpFormDetails.email"
          label={
                    Object {
                              "content": <Icon
                                as="i"
                                color="blue"
                                name="envelope"
                                size="small" />,
                            }
          }
          labelPosition="left"
          name="email"
          placeholder="Email"
          type="email"
          value="" />,
        <label
          target="signup.signUpFormDetails.phoneNumber">
          Phone Number
</label>,
        <Field
          component={[Function]}
          id="signup.signUpFormDetails.phoneNumber"
          label={
                    Object {
                              "content": <Icon
                                as="i"
                                color="blue"
                                name="phone"
                                size="small" />,
                            }
          }
          labelPosition="left"
          name="phoneNumber"
          placeholder="Phone Number"
          type="number"
          value="" />,
      ],
      "component": "div",
      "name": "signUpFormDetails",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "First Name",
          "target": "signup.signUpFormDetails.firstName",
        },
        "ref": null,
        "rendered": "First Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "component": [Function],
          "id": "signup.signUpFormDetails.firstName",
          "label": Object {
            "content": <Icon
              as="i"
              color="blue"
              name="user"
              size="small" />,
          },
          "labelPosition": "left",
          "name": "firstName",
          "placeholder": "First Name",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Last Name",
          "target": "signup.signUpFormDetails.lastName",
        },
        "ref": null,
        "rendered": "Last Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "id": "signup.signUpFormDetails.lastName",
          "label": Object {
            "content": <Icon
              as="i"
              color="blue"
              name="user"
              size="small" />,
          },
          "labelPosition": "left",
          "name": "lastName",
          "placeholder": "Last Name",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Password",
          "target": "signup.signUpFormDetails.password",
        },
        "ref": null,
        "rendered": "Password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "id": "signup.signUpFormDetails.password",
          "label": Object {
            "content": <Icon
              as="i"
              color="blue"
              name="lock"
              size="small" />,
          },
          "labelPosition": "left",
          "name": "Password",
          "placeholder": "Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email",
          "target": "signup.signUpFormDetails.email",
        },
        "ref": null,
        "rendered": "Email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "id": "signup.signUpFormDetails.email",
          "label": Object {
            "content": <Icon
              as="i"
              color="blue"
              name="envelope"
              size="small" />,
          },
          "labelPosition": "left",
          "name": "email",
          "placeholder": "Email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Phone Number",
          "target": "signup.signUpFormDetails.phoneNumber",
        },
        "ref": null,
        "rendered": "Phone Number",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "id": "signup.signUpFormDetails.phoneNumber",
          "label": Object {
            "content": <Icon
              as="i"
              color="blue"
              name="phone"
              size="small" />,
          },
          "labelPosition": "left",
          "name": "phoneNumber",
          "placeholder": "Phone Number",
          "type": "number",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label
            target="signup.signUpFormDetails.firstName">
            First Name
</label>,
          <Field
            autoFocus={true}
            component={[Function]}
            id="signup.signUpFormDetails.firstName"
            label={
                        Object {
                                    "content": <Icon
                                      as="i"
                                      color="blue"
                                      name="user"
                                      size="small" />,
                                  }
            }
            labelPosition="left"
            name="firstName"
            placeholder="First Name"
            value={undefined} />,
          <label
            target="signup.signUpFormDetails.lastName">
            Last Name
</label>,
          <Field
            component={[Function]}
            id="signup.signUpFormDetails.lastName"
            label={
                        Object {
                                    "content": <Icon
                                      as="i"
                                      color="blue"
                                      name="user"
                                      size="small" />,
                                  }
            }
            labelPosition="left"
            name="lastName"
            placeholder="Last Name"
            value={undefined} />,
          <label
            target="signup.signUpFormDetails.password">
            Password
</label>,
          <Field
            component={[Function]}
            id="signup.signUpFormDetails.password"
            label={
                        Object {
                                    "content": <Icon
                                      as="i"
                                      color="blue"
                                      name="lock"
                                      size="small" />,
                                  }
            }
            labelPosition="left"
            name="Password"
            placeholder="Password"
            type="password"
            value="" />,
          <label
            target="signup.signUpFormDetails.email">
            Email
</label>,
          <Field
            component={[Function]}
            id="signup.signUpFormDetails.email"
            label={
                        Object {
                                    "content": <Icon
                                      as="i"
                                      color="blue"
                                      name="envelope"
                                      size="small" />,
                                  }
            }
            labelPosition="left"
            name="email"
            placeholder="Email"
            type="email"
            value="" />,
          <label
            target="signup.signUpFormDetails.phoneNumber">
            Phone Number
</label>,
          <Field
            component={[Function]}
            id="signup.signUpFormDetails.phoneNumber"
            label={
                        Object {
                                    "content": <Icon
                                      as="i"
                                      color="blue"
                                      name="phone"
                                      size="small" />,
                                  }
            }
            labelPosition="left"
            name="phoneNumber"
            placeholder="Phone Number"
            type="number"
            value="" />,
        ],
        "component": "div",
        "name": "signUpFormDetails",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "First Name",
            "target": "signup.signUpFormDetails.firstName",
          },
          "ref": null,
          "rendered": "First Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "component": [Function],
            "id": "signup.signUpFormDetails.firstName",
            "label": Object {
              "content": <Icon
                as="i"
                color="blue"
                name="user"
                size="small" />,
            },
            "labelPosition": "left",
            "name": "firstName",
            "placeholder": "First Name",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Last Name",
            "target": "signup.signUpFormDetails.lastName",
          },
          "ref": null,
          "rendered": "Last Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "id": "signup.signUpFormDetails.lastName",
            "label": Object {
              "content": <Icon
                as="i"
                color="blue"
                name="user"
                size="small" />,
            },
            "labelPosition": "left",
            "name": "lastName",
            "placeholder": "Last Name",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password",
            "target": "signup.signUpFormDetails.password",
          },
          "ref": null,
          "rendered": "Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "id": "signup.signUpFormDetails.password",
            "label": Object {
              "content": <Icon
                as="i"
                color="blue"
                name="lock"
                size="small" />,
            },
            "labelPosition": "left",
            "name": "Password",
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email",
            "target": "signup.signUpFormDetails.email",
          },
          "ref": null,
          "rendered": "Email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "id": "signup.signUpFormDetails.email",
            "label": Object {
              "content": <Icon
                as="i"
                color="blue"
                name="envelope"
                size="small" />,
            },
            "labelPosition": "left",
            "name": "email",
            "placeholder": "Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Phone Number",
            "target": "signup.signUpFormDetails.phoneNumber",
          },
          "ref": null,
          "rendered": "Phone Number",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "id": "signup.signUpFormDetails.phoneNumber",
            "label": Object {
              "content": <Icon
                as="i"
                color="blue"
                name="phone"
                size="small" />,
            },
            "labelPosition": "left",
            "name": "phoneNumber",
            "placeholder": "Phone Number",
            "type": "number",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
